### ISSUE TITLE
Unable to set target with a name other than "map"

### ISSUE THREAD

**Issue by clintsinger** [openlayers/openlayers#9383]
**Describe the bug**
I am working on an angular component where I am hosting a map control.  
In the HTML I have a div with the id of 'map' and then in my code I set the target to 'map' and everything works.  If I change the id of the div and set the target to any other name the map no longer renders.  If I change it all back to 'map' again it goes back to working.

**To Reproduce**
1. Create a div with some id other than 'map', for example 'pam'
2. Set the map target to 'pam'
3. Observe that the map doesn't render.

**Expected behavior**
I expect to be able to pick any name for the id and be able to set the target to that name.  The goal here is to have unique names so I can host multiple maps on the same page; which can't be done if they all point to the same 'map' div.

**Comment by fredj** [comment]
See https://codesandbox.io/s/xp5pzl17p4
Here the map div id is `pam` (in index.html and index.js) and the map renders as expected.

**Comment by clintsinger** [comment]
Well, at least it isn't an obvious issue.  Perhaps it has something with being hosted in an Angular 7 component.  I could certainly see it not working at all which is why my situation is bizarre because it does work as long as I use 'map' as the id and target.

Looks like I have to do some more investigating.

**Comment by clintsinger** [comment]
I tried initializing the map outside of Angular using the `NgZone.runOutsideAngular()` method and it looks like it is working now.   

I am still baffled though how it worked when using 'map' as the id and target but nothing else until I made that change.   I had only used map because that was the default in all of the examples and it made sense to call it that.  I guess if I had tried something else from the beginning I would have been even more frustrated because then I wouldn't have seen it work ever and at least in this case I had hope that it would work :)
